### DISCUSSION TITLE
Does TF-controller bloc any manual change ?

### DISCUSSION THREAD

**Discussion by mehdi-wsc** [flux-iac/tofu-controller#459] [Q&A]
Hello Folks,
Currently I'm trying to test TF-controller and compare it to crossplane ( another solution for IaC via gitops ).
The aim of using Gitops is to stop and avoid all manual changes, I tried this with tf-controller but it doesn't seem the case, you can modify the resources via aws console after creation and when this occurs the Terraform instance in kubernetes will be in error loop.
Can we add any feature to stop the manual changes ? So if someone want to change something manually , it will not be considered because we have one source of deployment via the controller. ( This how crossplan works but without terraform, so if tf-controller can leverage between controlling manual changes and usage of terraform will be the best tool ever)
Thank you !

**Comment by chanwit** [comment]
Yes we already have that implemented via our drift detection.

If you set approvePlan to "auto", all of your resources would be reverted back to the good state.

Many users are using this feature on production clusters instead of Crossplane.
